### PR TITLE
Add TypeScript typings

### PR DESCRIPTION
**Pull request by eramdam**
I needed to use meyda in a TypeScript project and needed typings, so I wrote them 😅

### CODE REVIEW

**Comment by hughrawlinson** [comment]
Woohoo! Thanks @eramdam!

@nevosegal @jakubfiala @2xAA @odub have you worked with typescript? If so, better you review than I.

**Comment by nevosegal** [comment]
I don't I'm afraid :/

**Comment by nevosegal** [comment]
I don't I'm afraid :/

**Comment by hughrawlinson** [comment]
@eramdam Hi, sorry for such a long delay here :disappointed:. It's always hard for me to find time to work on open source stuff.

I'm still not super familiar with Typescript, but I wonder what the ramifications would be of choosing to put these typings in the [Definitely Typed] repository instead? Is that repo still widely used, or do typescripters prefer to have types built into the repos they depend on directly?

[Definitely Typed]: https://github.com/DefinitelyTyped/DefinitelyTyped

**Comment by eramdam** [comment]
@hughrawlinson I guess the only difference would be that the types can be updated independentely of the library they map too. If that's easier for you folks I can put those types there instead :)

**Comment by hughrawlinson** [comment]
@eramdam I think I would prefer that, mostly because I don't know enough typescript to maintain types in this repo.

**Comment by eramdam** [comment]
@hughrawlinson That's fair! I'll do that when I have the time then :)

**Comment by hughrawlinson** [comment]
Awesome! Let us know if you do it! :smile: Thanks so much for taking the time to open the PR

**Comment by hughrawlinson** [comment]
Hey @eramdam - I'm regretting not merging this PR, because I need ts types now for Meyda 😛 Mind if I reopen it and merge it? Or did you end up putting this in a types repo I'm not aware of?

File: meyda.d.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  export class MeydaAnalyzer {
+    start(features: Array<MeydaAudioFeature>): void;
+
+    stop(): void;
+
+    setSource(source: MeydaSource): void;
+
+    get(features: Array<MeydaAudioFeature>): Partial<MeydaFeaturesObject> | null;
```
Comment (hughrawlinson) [inline meyda.d.ts]: OMG I didn't know about Partial, this is so good!!! Thanks!

**Comment by hughrawlinson** [comment]
I take it back, [the Typescript publishing docs] say that if the module itself is not written in typescript, the type files should go in the `@types` org on npm. It seems like that can be done by making a PR to [this project](https://definitelytyped.org/). If you'd like to do that, please go for it! Otherwise, I'm happy to pursue it :)

**Comment by eramdam** [comment]
> I take it back, [the Typescript publishing docs] say that if the module itself is not written in typescript, the type files should go in the `@types` org on npm. It seems like that can be done by making a PR to [this project](https://definitelytyped.org/). If you'd like to do that, please go for it! Otherwise, I'm happy to pursue it :)

Oof, completely forgot about that 🙈 I guess if you're in the mood of opening a PR there, I won't stop you from doing so 😁 Also i don't know if the codebase changed since I opened the PR so i don't know if the types are accurate still (I haven't worked on the project that uses those typings in a while)

**Comment by hughrawlinson** [comment]
Cool - as long as you don't mind me opening a PR with your types (plus a few minor changes), I'll go ahead with that.

**Comment by eramdam** [comment]
> Cool - as long as you don't mind me opening a PR with your types (plus a few minor changes), I'll go ahead with that.

I'm ok with that!

**Comment by hughrawlinson** [comment]
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41411 for progress 😄

**Comment by eramdam** [comment]
Closing this PR since there's one on DT now :)

**Pull request closed without merging**